miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppManifest.kt

```diff
@@ -0,0 +1,19 @@
+package com.rakuten.tech.mobile.miniapp
+
+import androidx.annotation.Keep
+import com.rakuten.tech.mobile.miniapp.permission.MiniAppCustomPermissionType
+
+/**
+ * A data class to represent data in the mini app's manifest.
+ */
+@Keep
+data class MiniAppManifest(
+    // List of permissions requested by Mini App in their manifest
+    val requiredPermissions: List<Pair<MiniAppCustomPermissionType, String>>,
+    val optionalPermissions: List<Pair<MiniAppCustomPermissionType, String>>,
+    // custom metadata set by Mini App in their manifest
+    private val customMetaData: Map<String, String>
+) {
+    // get metadata value of "customMetaData" per key set by Host App
+    fun getMetadataValue(key: String) = customMetaData[key]
```

Comment:
Sorry, I think there was something like this in the draft specs, but it seems like we don't really need it now since we changed to using a specific key for custom meta data. 

Probably you can remove this and change `customMetaData` to public. Since you're using `Map` as the type it will already be immutable.